services/request_service.go

```diff
@@ -30,137 +29,125 @@ func NewRequestService(didMethod string, ledgerService LedgerServiceI) RequestSe
 	}
 }
 
-func (rs RequestService) IsDidUrl(didUrl string) bool {
-	_, path, query, fragmentId, err := cheqdUtils.TrySplitDIDUrl(didUrl)
-	return err == nil && (path != "" || query != "" || fragmentId != "")
-}
-
-func (rs RequestService) ProcessDIDRequest(didUrl string, resolutionOptions types.ResolutionOption) (string, error) {
+func (rs RequestService) ProcessDIDRequest(didUrl string, resolutionOptions types.ResolutionOption) (string, int) {
 	var result string
-	var err error
-	if rs.IsDidUrl(didUrl) {
+	var statusCode int
+	if utils.IsDidUrl(didUrl) {
 		log.Trace().Msgf("Dereferencing %s", didUrl)
-		result, err = rs.prepareDereferencingResult(didUrl, types.DereferencingOption(resolutionOptions))
+		result, statusCode = rs.prepareDereferencingResult(didUrl, types.DereferencingOption(resolutionOptions))
 	} else {
 		log.Trace().Msgf("Resolving %s", didUrl)
-		result, err = rs.prepareResolutionResult(didUrl, resolutionOptions)
-	}
-
-	if resolutionOptions.Accept == types.HTML {
-		return "<!DOCTYPE html><html><body><h1>Cheqd DID Resolver</h1><pre id=\"r\"></pre><script> var data = " +
-			result + ";document.getElementById(\"r\").innerHTML = JSON.stringify(data, null, 4);" +
-			"</script></body></html>", err
+		result, statusCode = rs.prepareResolutionResult(didUrl, resolutionOptions)
 	}
-	return result, err
+	return result, statusCode
 }
 
-func (rs RequestService) prepareResolutionResult(did string, resolutionOptions types.ResolutionOption) (string, error) {
-	didResolution, err := rs.Resolve(did, resolutionOptions)
-	if err != nil {
-		return "", err
-	}
+func (rs RequestService) prepareResolutionResult(did string, resolutionOptions types.ResolutionOption) (string, int) {
+	didResolution := rs.Resolve(did, resolutionOptions)
 
-	resolutionMetadata, err := json.Marshal(didResolution.ResolutionMetadata)
-	if err != nil {
-		return "", err
-	}
-
-	didDoc, err := rs.didDocService.MarshallDID(didResolution.Did)
-	if err != nil {
-		return "", err
-	}
-
-	metadata, err := json.Marshal(&didResolution.Metadata)
-	if err != nil {
-		return "", err
+	resolutionMetadata, mErr1 := json.Marshal(didResolution.ResolutionMetadata)
+	didDoc, mErr2 := rs.didDocService.MarshallDID(didResolution.Did)
+	metadata, mErr3 := json.Marshal(&didResolution.Metadata)
+	if mErr1 != nil || mErr2 != nil || mErr3 != nil {
+		return createJsonResolutionInternalError(resolutionMetadata)
```

Comment:
Can we pass the actual error inside?

Reply:
No, we can't. Only `internalError` message is available for Resolution Metadata